submissions/DmytryjK/task_js_oop/js/app.js

```diff
@@ -0,0 +1,158 @@
+    }
+    update(dt) {    
+        this.x += this.speed * dt;
+        this.collisionСheck();
+        this.addingRandomSpeed();
+    }
+    render() {
+        ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+    }
+    collisionСheck() {
+        if(this.player.y == this.y && this.player.x <= Math.floor(this.x) + this.player.heroWidth / 1.5 && this.player.x >= Math.floor(this.x) - this.player.heroWidth / 1.5 ) {
+            setTimeout(deletePopupMessage, 1000, 'body', 'popUp');
+            showPopupMessage('div','popUp', 'body', 'You losе, bro :(');
+            resetPlayerPosition(player);
+            scoreCount = 0;
+            score(scoreCount);
+        }
+    }
+    addingRandomSpeed() {
+        if (this.x > this.fieldWidth + this.enemyWidth) {
+            this.x = -this.enemyWidth;
+            this.speed = baseConfig.speedRandom();
+        }
+    }
+};
+
+class Player {
+    constructor(fieldWidth, fieldHeight, sprite, heroWidth, heroHeight) {
+        this.x = fieldWidth / 2;
+        this.y = fieldHeight;
+        this.sprite = sprite;
+        this.fieldWidth = fieldWidth;
+        this.fieldHeight = fieldHeight;
+        this.heroWidth = heroWidth;
+        this.heroHeight = heroHeight;
+    }
+    update(dt) {
+        this.outOfFiledCheck();
+        this.checkingForVictory();
+    }
+    render() {
+        ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+    }
+    handleInput(key) {
+        switch(key) {
+            case "left":
+                this.x += -this.heroWidth; 
+                break;
+            case "up":
+                this.y += -this.heroHeight/2; 
+                break;
+            case "right":
+                this.x += this.heroWidth; 
+                break;
+            case "down":
+                this.y += this.heroHeight/2;
+                break;
+        }
+    }
+    outOfFiledCheck() {
+        if (this.x < 0 || 
+            this.x > this.fieldWidth || 
+            this.y > this.fieldHeight) {
+                resetPlayerPosition(player);
+        } 
+    }
+    checkingForVictory() {
+        if (this.y <= 0) {
+            setTimeout(deletePopupMessage, 1000, 'body', 'popUp');
+            showPopupMessage('div','popUp', 'body', 'You win! :)');
+            resetPlayerPosition(player); 
+            scoreCount ++;
+            score();
+        }
+    }
+};
+
+const player = new Player(baseConfig.fieldWidth, baseConfig.fieldHeight, baseConfig.playerLinkImg, baseConfig.heroWidth, baseConfig.heroHeight);
+
+const   firstEnemy  = new Enemy(0, 50, baseConfig.speedRandom(), baseConfig.enemyLinkImg, baseConfig.fieldWidth, baseConfig.enemyWidth, player),
+        secondEnemy  = new Enemy(0, 135, baseConfig.speedRandom(), baseConfig.enemyLinkImg, baseConfig.fieldWidth, baseConfig.enemyWidth, player),
+        thirdEnemy  = new Enemy(0, 220, baseConfig.speedRandom(), baseConfig.enemyLinkImg, baseConfig.fieldWidth, baseConfig.enemyWidth, player);
```

Comment:
What are 50, 135 and 220?
As a peer developer I'd like to have a semantical meaning of magic numbers to be able to add more enemies to the gameplay.
Also can we please have here a less verbose code to add enemies? Say, we need 20 more.

Reply:
fixed

Reply:
Very cool advice, I don’t understand how I didn’t notice this)

Reply:
Hint: Also I do not want to count pixels. I want to add an enemy knowing the number of the row on the playfield

Reply:
hmm, I added cycle for, and you have to add only numbers of enemies to create the array, without an option to select field number)
would that be ok, or does it have to be a field selection?

Reply:
I want to add an enemy that will run along row 2 of the field. What do i do? How can I do this in the least painful way without trial-and-error and without seeking an advice from the project author that may have already quit the company by now?

Reply:
I did, it seems to work correctly)